app/models/package_manager/go.rb

```diff
@@ -34,5 +48,22 @@ def self.mapping(project)
         repository_url: "https://#{project['Package']}"
       }
     end
+
+    # https://golang.org/cmd/go/#hdr-Import_path_syntax
+    def self.resolved_name(name)
+      return name if name.start_with?(*KNOWN_HOSTS)
+      return name if KNOWN_VCS.any?(&name.method(:include?))
+
+      go_import = get_html('https://' + name + '?go-get=1')
```

Comment:
Could we add a new column to project to store resolved names or something like that? I'm conflicted about what to suggest on this PR. One of the worries I have is how long that redirect could take since we are making a call out to the internet here.

Reply:
We could definitely add that column. Would we then have to worry about flushing it if a redirecting name goes away?

Reply:
Eh, that's a good point. I'm not too familiar with Go and how often this stuff would occur or even how this naming thing is used in the community. It might be something we could handle in the `PackageManager::Go.update` method?